packages/react-dom/src/__tests__/ReactDOMServerIntegrationUntrustedURL-test.js

```diff
@@ -327,9 +335,14 @@ describe('ReactDOMServerIntegration - Untrusted URLs - disableJavaScriptURLs', (
 
   it('rejects a javascript protocol href if it is added during an update', () => {
     const container = document.createElement('div');
-    ReactDOM.render(<a href="http://thisisfine/">click me</a>, container);
-    expect(container.firstChild.href).toBe('http://thisisfine/');
-    ReactDOM.render(<a href="javascript:notfine">click me</a>, container);
+    const root = ReactDOMClient.createRoot(container);
+    ReactDOM.flushSync(() => {
+      root.render(<a href="thisisfine">click me</a>);
+    });
+    expect(container.firstChild.href).toBe('http://localhost/thisisfine');
```

Comment:
This is unexpected and seems like a breaking change in 19? 

https://codesandbox.io/p/sandbox/beautiful-tereshkova-lklwsl

Reply:
PR to isolate this: https://github.com/facebook/react/pull/28137